File: single_player/easy/c/the_descent.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

/**
 * The idea is to fire at the tallest of the scanned mountains
 * when the ships reaches it.
 */
int main()
{
    while (1) {
        // Ship position
        int spaceX;
        int spaceY;
        scanf("%d%d", &spaceX, &spaceY);

        // Mountain data
        int mountains[8] = {0};
        int tallestMountainLocation = 0;
        int tallestMountainHeight = 0;

        for (int i = 0; i < 8; i++) {
            // This mountain's height
            int mountainH;
            scanf("%d", &mountainH);

            // Check if it's the tallest known mountain
            if (mountainH > tallestMountainHeight)
            {
                tallestMountainHeight = mountainH;
                tallestMountainLocation = i;
            }
        }

        // Fire only when at tallest mountain's X position
        printf((spaceX == tallestMountainLocation) ? "FIRE\n" : "HOLD\n");
    }

    return 0;
}

File: single_player/easy/js/the_descent.js
for (;;) {
    var inputs = readline().split(' '),
        SX = +inputs[0],    // Current ship X position
        SY = +inputs[1],    // Current ship Y position
        M = [],     // Array of mountain heights
        MH = 0,     // Mountain height
        MHmax = 0;  // Tallest mountain (maximum mountain height)

    // Find tallest mountain in game
    for (var i = 0; i < 8; i++) {
        MH = +readline();
        if (MHmax < MH) MHmax = MH;
        M.push(MH);
    }

    // If passing over tallest mountain, target it
    print((MHmax === M[SX]) ? 'FIRE' : 'HOLD');
}
